vspace/src/main/java/edu/asu/diging/vspace/web/staff/SpaceOverviewController.java

```diff
@@ -0,0 +1,317 @@
+                                && link.getLink().getTarget().getId() != null) {
+                            spaceAllLink.add(link.getLink().getTarget().getId());
+                        }
+                    });
+                }
+                if (space.getId() != null) {
+                    spaceLinkMap.put(space.getId(), spaceAllLink);
+                }
+            }
+        }
+    }
+
+    /**
+     * To map each space to all linked space or each space to all linked module
+     * 
+     * @param <T>
+     * @param spaceToAllOtherLinksList passing list of space to space link or space
+     *                                 to module
+     * @param spaceToAllOtherLinksMap  To hold list of space to space link or space
+     *                                 to module corresponding to each space
+     * @param isModule                 boolean variable to identify whether it is
+     *                                 module or not
+     */
+    private <T> void getSpaceToAllOtherLinks(Iterable<T> spaceToAllOtherLinksList,
+            Map<String, List<T>> spaceToAllOtherLinksMap, boolean isModule) {
+
+        for (T link : spaceToAllOtherLinksList) {
+            String spaceId = null;
+            if (isModule) {
+                IModuleLinkDisplay moduleLink = (IModuleLinkDisplay) link;
+
+                if (moduleLink.getLink() != null && moduleLink.getLink().getSpace() != null
+                        && moduleLink.getLink().getSpace().getId() != null) {
+
+                    spaceId = moduleLink.getLink().getSpace().getId();
+                }
+            } else {
+                ISpaceLinkDisplay spaceLink = (ISpaceLinkDisplay) link;
+
+                if (spaceLink.getLink() != null && spaceLink.getLink().getSourceSpace() != null
+                        && spaceLink.getLink().getSourceSpace().getId() != null) {
+
+                    spaceId = spaceLink.getLink().getSourceSpace().getId();
+                }
+            }
+            if (spaceId != null) {
+                if (!spaceToAllOtherLinksMap.containsKey(spaceId)) {
+                    spaceToAllOtherLinksMap.put(spaceId, new ArrayList<>());
+                }
+                spaceToAllOtherLinksMap.get(spaceId).add(link);
+            }
+        }
+    }
+
+    /**
+     * creating json for space and module node in the spaceoverview graph
+     * 
+     * @param <T>
+     * @param contextPath This variable holds the contextpath of the application
+     * @param nodeList    List of space/module
+     * @param mapper      For creating ObjectNode
+     * @param nodeArray   Creating Array of ObjectNode for nodes in spaceoverview
+     *                    graph
+     * @param allNodeList List of all the nodes of spaceoverview graph
+     * @param isModule    boolean variable to indicate whether it is module or not
+     */
+    private <T> void createNodeForOverviewGraph(String contextPath, Iterable<T> nodeList, ObjectMapper mapper,
```

Comment:
instead of the generics that don't really do anything for you here, just pass `Iterable<? extends IVSpaceElement>` and get rid of the `isModule` flag. Then iterate over `IVSpaceElement` and check with `instanceOf`.